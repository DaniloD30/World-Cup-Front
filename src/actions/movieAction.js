import movieService from "../services/movieService";
import Constants from "../helpers/enums/Constants";
import Utils from "../helpers/utils";

const getMovies = (LOADING_IDENTIFICATOR = "", callback = () => {}) => (
  dispatch
) => {
  dispatch(Utils.startLoading(LOADING_IDENTIFICATOR));

  return movieService
    .getMovies()
    .then((response) => {
      if (response) {
        // Utils.setToken(response.data.Token);
        // let jwt = jwtDecode(response.data.Token);
        callback();
        dispatch({
          type: Constants.GET_ALL,
          payload: response.data,
        });
      }
    })
    .catch((error) => {
      callback(error);
    })
    .finally(() => {
      dispatch(Utils.endLoading(LOADING_IDENTIFICATOR));
    });
};

const generateChampionship = (
  values,
  LOADING_IDENTIFICATOR = "",
  callback = () => {}
) => (dispatch) => {
  dispatch(Utils.startLoading(LOADING_IDENTIFICATOR));

  return movieService
    .getWinner(values)
    .then((response) => {
      if (response) {
        callback();
        dispatch({
          type: Constants.WINNERS,
          payload: response.data,
        });
      }
    })
    .catch((error) => {
      callback(error);
    })
    .finally(() => {
      dispatch(Utils.endLoading(LOADING_IDENTIFICATOR));
    });
};

export const saveLength = (length) => ({
  type: Constants.SAVE_LIST_MOVIES,
  payload: length,
});

export default {
  getMovies,
  generateChampionship,
  saveLength,
};
