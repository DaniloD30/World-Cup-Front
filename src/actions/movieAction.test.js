import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import movieAction from "./movieAction";
export const mockStore = configureMockStore([thunk]);
const store = mockStore();

describe("actions should working to store", () => {
  beforeEach(() => {
    // Runs before each test in the suite
    store.clearActions();
  });

  describe("movieAction", () => {
    test("Dispatches the getMovies action and payload", () => {
      store.dispatch(movieAction.getMovies("identificatorTest"));
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
