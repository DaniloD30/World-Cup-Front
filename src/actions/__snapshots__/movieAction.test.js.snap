// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should working to store movieAction Dispatches the getMovies action and payload 1`] = `
Array [
  Object {
    "loading": Object {
      "identificatorTest": true,
    },
    "type": "UPDATE_LOADING",
  },
]
`;
